lib/sidekiq/api.rb

```diff
@@ -242,11 +242,19 @@ def size
       Sidekiq.redis { |con| con.llen(@rname) }
     end
 
-    # Sidekiq Pro overrides this
+    # Sidekiq Pro overrides pause?, pause! and unpause! functions
     def paused?
       false
     end
 
+    def pause!
```

Comment:
What happens if the user isn't using Sidekiq Pro?  Should these methods really do nothing?  Should they raise an error or log a warning?  Should the button be disabled in the Web UI with an explanatory tooltip if `!defined?(::Sidekiq::Pro)`?

Reply:
That's good question, I left it under open question so we could discuss some possible options. I am currently of the opinion that we roll it out with minimal change to the regular sidekiq and web (backwards compatible) and as a feature on sidekiq pro. That means, we simply hide the button when pro is not present from the UI and accordingly add proper validation in the api router. Raising exception or tooltips would result in unnecessary friction that didn't exist before. Of course, can be persuaded :). 

I introduced this as a new commit. Its not a huge change. Used `defined?(::Sidekiq::Pro)` as a check to determine whether or not pro is present.